### PR TITLE
[MO][IR READER] Updated MO and IR Reader to keep the second MaxPool output in restored IR

### PR DESCRIPTION
**Pull request by yekruglov**
Root cause analysis: `MaxPool` operation in restored IR has only one output port. It happens, because IR Reader does not apply back transformations by default. Also `AvgPool` in restored IR has `MaxPool` specified attributes: `axis`, `dilations` and `index_element_type`.

Solution: Apply `Maxpool` `normalize_outputs` transformation in `IR Reader` and update `Pooling` `backend_attrs` to keep correct attributes

Ticket: CVS-73712

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests N/A
* [x]  Framework operation tests N/A
* [x]  Transformation tests N/A
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list N/A
* [x]  Guide on how to convert the **public** model N/A
* [x]  User guide update N/A

### CODE REVIEW

File: tools/mo/openvino/tools/mo/back/MaxPool.py [hunk 1]
```diff
@@ -39,4 +43,4 @@ def replace_pattern(self, graph: Graph, match: dict):
         if node.out_port(1).disconnected():
             output = Result(node.graph, {'name': node.name + '/Result_port_1/',
                                          'keep_output_port': node.has_and_set('remove_values_output')}).create_node()
-            node.out_port(1).get_connection().set_destination(output.in_port(0))
+            node.out_port(1).get_connection().set_destination(output.in_port(0))
```
Comment (achetver) [inline tools/mo/openvino/tools/mo/back/MaxPool.py:46]: nit: add a newline at EOF.

**Comment by achetver** [comment]
In general looks good. @yekruglov As I understand you checked these changes with both Max- and AvgPool operations?

**Review by achetver — APPROVED** [review]

File: tools/mo/openvino/tools/mo/ops/pooling.py [hunk 2]
```diff
@@ -91,14 +91,18 @@ def backend_attrs(self):
...
+            ('auto_pad', lambda node: node.auto_pad if node.has_valid('auto_pad') else 'explicit')
         ]
 
+        if self.attrs.get('pool_method') == 'avg':
+            return backend_attrs_list
+        else:
+            return backend_attrs_list + [
+                ('dilations', lambda node: ','.join(map(str, node['dilation'][node.spatial_dims]))),
+                'axis',
+                ('index_element_type', lambda node: np_data_type_to_destination_type(node.index_element_type))
```
Comment (pavel-esir) [inline tools/mo/openvino/tools/mo/ops/pooling.py:103]: There is missing `('auto_pad', lambda node: node.auto_pad if node.has_valid('auto_pad') else 'explicit')`
Is it expected?

**Review by pavel-esir — APPROVED** [review]
looks good to me.
nit: please use CVS- ticket number prefix. This will enable integration between Jira and github

**Review by mvafin — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Pull request merged**